sklearn/cross_decomposition/pls_.py

```diff
@@ -271,6 +271,10 @@ def fit(self, X, Y):
 
         # NIPALS algo: outer loop, over components
         for k in range(self.n_components):
+            if np.all(np.dot(Yk.T, Yk) < np.finfo(np.double).eps):
+                # Yk constant
+                warnings.warn('Y constant at iteration %s' % k)
+                break
```

Comment:
I am not familiar with this code nor CCA and PLS in general. Can you explain what this means? Would it be possible to write a unit test to specifically check this behavior?